File: framework/src/gmsec/internal/StatusException.cpp

/*
 * Copyright 2007-2021 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */







/**
 *  @file StatusException.cpp
 *
 *  This file contains the StatusException class.
*/

#include <gmsec/Status.h>
#include <gmsec/internal/StatusException.h>

using gmsec::Status;
using gmsec::internal::StatusException;

StatusException::StatusException() : Status()
{
}

StatusException::StatusException(GMSEC_STATUS_CLASS eclass, unsigned int code, const char *text) : Status(eclass,code,text)
{
}

StatusException::StatusException(const StatusException &st) : Status(st)
{
}

File: java/gov/nasa/gsfc/gmsecapi/internal/StatusException.java
/*
 * Copyright 2007-2021 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */







package gov.nasa.gsfc.gmsecapi.internal;

import gov.nasa.gsfc.gmsecapi.Status;


/**
 * @class StatusException
 *
 * @brief This exception can be thrown by Connect constructors,
 * but is hidden by ConnectionFactory::Create().
 * 
 */
public class StatusException extends Exception
{
	private static final long serialVersionUID = 1L;

	private Status status = null;

	public StatusException(Status status)
	{
		this.status = status;
	}

	public Status getStatus()
	{
		return status;
	}

}
